hf_vision_model_tfserving_gke/deploy.sh

```diff
@@ -0,0 +1,119 @@
+#! /bin/sh
+# This shell script apply Deployment to the designated GKE cluster
+# Prerequisite: You should provision GKE cluster beforehand
+
+## 0. Define Variables
+GCP_PROJECT_ID=gcp-ml-172005
+### Path where service account key file is (JSON)
+SA_KEY_FILE=/tmp/gcp-ml-172005-5c6ccb3685b9.json
+
+GKE_CLUSTER_NAME=cluster-1
+GKE_CLUSTER_ZONE=us-central1-c
+
+### Base TFServing Docker image
+BASE_IMAGE_TAG=gcr.io/gcp-ml-172005/tfs-resnet-cpu-opt
+
+### Repository where to download SavedModel from
+MODEL_RELEASE_REPO=deep-diver/deploy-hf-tf-vision-models
+MODEL_RELEASE_TAG=1.0
+MODEL_RELEASE_FILE=saved_model.tar.gz
+
+### Model name to be embeded into the TFServing 
+MODEL_NAME=hf-vit
+### VERSION which will be added at the end of Docker Image tag
+VERSION=latest
+
+GKE_DEPLOYMENT_NAME=tfs-server
+TARGET_EXPERIMENT="8vCPU+16GB+inter_op4"
+
+### -z is a testing functionality
+if [ -z ${GCP_PROJECT_ID} ] || \
+   [ -z ${SA_KEY_FILE} ] || \
+   [ -z ${GKE_CLUSTER_NAME} ] || \
+   [ -z ${GKE_CLUSTER_ZONE} ] ||  \
+   [ -z ${BASE_IMAGE_TAG} ] || \
+   [ -z ${MODEL_RELEASE_FILE} ] || \
+   [ -z ${MODEL_NAME} ] || \
+   [ -z ${VERSION} ] || \
+   [ -z ${GKE_DEPLOYMENT_NAME} ] || \
+   [ -z ${TARGET_EXPERIMENT} ]
+then
+    echo "Some variables in the shell script are not set. Check them with the below outputs"
+    echo "GCP_PROJECT_ID=$GCP_PROJECT_ID"
+    echo "SA_KEY_FILE=$SA_KEY_FILE"
+    echo "GKE_CLUSTER_NAME=$GKE_CLUSTER_NAME"
+    echo "GKE_CLUSTER_ZONE=$GKE_CLUSTER_ZONE"
+    echo "BASE_IMAGE_TAG=$BASE_IMAGE_TAG"
+    echo "MODEL_RELEASE_FILE=$MODEL_RELEASE_FILE"
+    echo "MODEL_NAME=$MODEL_NAME"
+    echo "VERSION=$VERSION"
+    echo "GKE_DEPLOYMENT_NAME=$GKE_DEPLOYMENT_NAME"
+    echo "TARGET_EXPERIMENT=$TARGET_EXPERIMENT"
+    exit 
+fi
+
+## 1. Download gcloud CLI tool
+ROOT_DIR=/tmp
+SAVED_FILENAME=gcloud_cli.tar.gz
+GCLOUD_DIRNAME=google-cloud-sdk
+curl https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-cli-393.0.0-linux-x86_64.tar.gz > $ROOT_DIR/$SAVED_FILENAME
+tar -xf $ROOT_DIR/$SAVED_FILENAME -C $ROOT_DIR
```

Comment:
I think in the README of this directory we could do the following:

* Write a section on installing `gcloud` and authentication. 
* Write a section on installing Docker and `kubectl`.
* Write a section on installing Kustomize.